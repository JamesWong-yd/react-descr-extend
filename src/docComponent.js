export default function docComponent(description, component, options = {}) {
  if (!component) {
    throw new Error('docComponent: component is required');
  }
  if (!description) {
    throw new Error('docComponent: description is required');
  }

  /* eslint-disable no-param-reassign */
  component.$$reactDesc = { description, options };
  /* eslint-enable no-param-reassign */
}
