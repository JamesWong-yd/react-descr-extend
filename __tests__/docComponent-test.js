import docComponent from '../src/docComponent';

class FakeComponent {}

it('fails for missing component propType', () => {
  expect(() => {
    docComponent(undefined);
  }).toThrowError('docComponent: component is required');
});

it('fails for missing description propType', () => {
  expect(() => {
    docComponent(undefined, FakeComponent);
  }).toThrowError('docComponent: description is required');
});

it('documents a basic component', () => {
  docComponent('component', FakeComponent);
  expect(FakeComponent.$$reactDesc).toBeDefined();
  expect(FakeComponent.$$reactDesc).toMatchSnapshot();
});

it('documents a deprecated component', () => {
  docComponent('component', FakeComponent, {
    deprecated: 'deprecated. use something else',
  });
  expect(FakeComponent.$$reactDesc).toBeDefined();
  expect(FakeComponent.$$reactDesc).toMatchSnapshot();
});
