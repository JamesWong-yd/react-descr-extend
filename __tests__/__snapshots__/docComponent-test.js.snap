exports[`test documents a basic component 1`] = `
Object {
  "description": "component",
  "options": Object {},
}
`;

exports[`test documents a deprecated component 1`] = `
Object {
  "description": "component",
  "options": Object {
    "deprecated": "deprecated. use something else",
  },
}
`;
